commons-numbers-arrays/src/main/java/org/apache/commons/numbers/arrays/SortInPlace.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.commons.numbers.arrays;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * Sort an array and perform the same reordering of entries on other arrays.
+ * For example, if
+ * <ul>
+ *  <li>{@code x = [3, 1, 2]}</li>
+ *  <li>{@code y = [1, 2, 3]}</li>
+ *  <li>{@code z = [0, 5, 7]}</li>
+ * </ul>
+ * then {@code Sort.ASCENDING.apply(x, y, z)} will update those arrays:
+ * <ul>
+ *  <li>{@code x = [1, 2, 3]}</li>
+ *  <li>{@code y = [2, 3, 1]}</li>
+ *  <li>{@code z = [5, 7, 0]}</li>
+ * </ul>
+ */
+public enum SortInPlace {
+    /** Sort in ascending order. */
+    ASCENDING((o1, o2) -> Double.compare(o1.key(), o2.key())),
+    /** Sort in descending order. */
+    DESCENDING((o1, o2) -> Double.compare(o2.key(), o1.key()));
+
+    /** Comparator. */
+    private final Comparator<PairDoubleInteger> comparator;
+
+    /**
+     * @param comparator Comparator.
+     */
+    SortInPlace(Comparator<PairDoubleInteger> comparator) {
+        this.comparator = comparator;
+    }
+
+    /**
+     * Sorts in place.
+     *
+     * @param x Array to be sorted and used as a pattern for permutation of
+     * the other arrays.
+     * @param yList Set of arrays whose permutations of entries will follow
+     * those performed on {@code x}.
+     * @throws IllegalArgumentException if not all arrays have the same size.
+     */
+    public void apply(double[] x,
+                      double[]... yList) {
+        final int yListLen = yList.length;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;yList&#39; (index: 1 | type: double[][). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;x&#39;. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).